client/auth/tokens.go

```diff
@@ -0,0 +1,40 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/SKF/go-utility/v2/auth/secretsmanagerauth"
+)
+
+type IdentityToken string
+
+type TokenProvider interface {
+	GetIdentityToken(ctx context.Context) (IdentityToken, error)
+}
+
+func (token IdentityToken) GetIdentityToken(ctx context.Context) (IdentityToken, error) {
+	return token, nil
+}
+
+func (token IdentityToken) String() string {
+	return string(token)
+}
+
+type SecretsManagerTokenProvider struct {
+	configured bool
+	Config     secretsmanagerauth.Config
+}
+
+func (provider *SecretsManagerTokenProvider) GetIdentityToken(ctx context.Context) (IdentityToken, error) {
+	if !provider.configured {
+		secretsmanagerauth.Configure(provider.Config)
+		provider.configured = true
+	}
+
+	if err := secretsmanagerauth.SignIn(ctx); err != nil {
```

Comment:
is it an improvement to handle 429 with a backoff?

Reply:
My initial plan was not to use go-utility/v2/auth here and do it using the client itself.

But I feel this is another PR